File: 219-contains-duplicate-ii/contains-duplicate-ii.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        num_dic = defaultdict(int)

        for i in range(len(nums)):
            if nums[i] in num_dic and i - num_dic[nums[i]] <= k:
                return True

            num_dic[nums[i]] = i
        
        return False

File: 219-contains-duplicate-ii/contains-duplicate-ii.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        Map<Integer, Integer> num_map = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int num = nums[i];
            
            if (num_map.containsKey(num) && i - num_map.get(num) <= k) {
 
                return true;
            }

            num_map.put(num, i);
        }

        return false;
    }
}